components/suggest/src/db.rs

```diff
@@ -128,7 +128,9 @@ impl<'a> SuggestDao<'a> {
         let (keyword_prefix, keyword_suffix) = split_keyword(&query.keyword);
         let suggestions_limit = query.limit.unwrap_or(-1);
 
-        let (mut statement, params) = if query.providers.contains(&SuggestionProvider::Pocket) {
+        let (mut statement, params) = if query.providers.contains(&SuggestionProvider::Pocket)
+            || query.providers.contains(&SuggestionProvider::Amo)
```

Comment:
Nit: you can also do this with the benifit that it will scan the provider array once.
```suggestion
        let (mut statement, params) = if query
            .providers
            .iter()
            .any(|p| matches!(p, SuggestionProvider::Pocket | SuggestionProvider::Amo))

```

Reply:
oooh fancy